### PR TITLE
Add user tests

### PR DESCRIPTION
**Pull request by jyx11011**
Add tests for user data manager, user json formatter
Add tests for user servlet doGet method.

### CODE REVIEW

File: src/main/java/com/google/step/model/User.java [hunk 1]
```diff
@@ -23,7 +23,11 @@ public User(long id, String email, String username, String photoBlobKey, String
     this.username = username;
     this.email = email;
     this.bio = bio;
-    this.photoBlobKey = Optional.of(photoBlobKey);
+    if (photoBlobKey != null) {
+      this.photoBlobKey = Optional.of(photoBlobKey);
+    } else {
+      this.photoBlobKey = null;
+    }
```
Comment (SuyashLakhotia) [inline src/main/java/com/google/step/model/User.java:30]: It feels weird to have three states - `null`, `Optional.absent()` and `Optional.of("some value")`.

Maybe you should just use `Optional.fromNullable()`, which will use `Optional.absent()` for null values?

File: src/test/java/com/google/step/datamanager/UserManagerDatastoreTest.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    Assert.assertEquals(EMAIL_A, user.email);
+    Assert.assertEquals(EMAIL_A, user.username);
+    Assert.assertEquals("", user.bio);
+    Assert.assertFalse(user.photoBlobKey.isPresent());
+  }
+
+  @Test
+  public void testReadUserByEmail_firstTime() {
+    User user = userManagerDatastore.readOrCreateUserByEmail(EMAIL_B);
+    Assert.assertEquals(EMAIL_B, user.email);
```
Comment (SuyashLakhotia) [inline src/test/java/com/google/step/datamanager/UserManagerDatastoreTest.java]: Just use a static import for `assertEquals` and you can get rid of `Assert.` everywhere in this file.

File: src/test/java/com/google/step/datamanager/UserManagerDatastoreTest.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    Assert.assertEquals(EMAIL_B, user.username);
+    Assert.assertEquals("", user.bio);
+    Assert.assertFalse(user.photoBlobKey.isPresent());
+  }
+
+  @Test
+  public void testReadUserByEmailExists() {
+    User user = userManagerDatastore.createUser(EMAIL_B);
```
Comment (SuyashLakhotia) [inline src/test/java/com/google/step/datamanager/UserManagerDatastoreTest.java:65]: You don't seem to be using `user` variable. Maybe just ignore the return value here and rename `userSecondTime` to `user`?

File: src/test/java/com/google/step/datamanager/UserManagerDatastoreTest.java [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    Assert.assertEquals(EMAIL_B, userSecondTime.email);
+    Assert.assertEquals(EMAIL_B, userSecondTime.username);
+    Assert.assertEquals("", userSecondTime.bio);
+    Assert.assertEquals(user.id, userSecondTime.id);
+    Assert.assertFalse(userSecondTime.photoBlobKey.isPresent());
+  }
+
+  @Test
+  public void testReadUserById_success() {
+    User userA = userManagerDatastore.createUser(EMAIL_A);
```
Comment (SuyashLakhotia) [inline src/test/java/com/google/step/datamanager/UserManagerDatastoreTest.java:76]: Similarly here.

**Review by SuyashLakhotia — CHANGES_REQUESTED** [review]
Nice tests! Do you mind cleaning up this PR so that the tests are the only changes? Seems like there are other changes as well?

File: src/test/java/com/google/step/datamanager/UserManagerDatastoreTest.java (hunk 3 above)
Comment (jyx11011) [reply src/test/java/com/google/step/datamanager/UserManagerDatastoreTest.java:65]: I use user.id below.

File: src/main/java/com/google/step/model/User.java (hunk 1 above)
Comment (jyx11011) [reply src/main/java/com/google/step/model/User.java:30]: In the update user method in user data manager, I pass in a User object with some null fields to update non-null filed and keep other null field unchanged.

**Review by SuyashLakhotia — DISMISSED** [review]

File: pom.xml [hunk 5]
```diff
@@ -35,6 +34,43 @@
...
+    <dependency>
+      <groupId>com.google.appengine</groupId>
+      <artifactId>appengine-api-stubs</artifactId>
+      <version>1.9.80</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.skyscreamer</groupId>
+      <artifactId>jsonassert</artifactId>
+      <version>1.5.0</version>
```
Comment (limli) [inline pom.xml:65]: add `<scope>test</scope>` ?

**Review by SuyashLakhotia — APPROVED** [review]

**Pull request merged**